src/modules/uxrce_dds_client/uxrce_dds_client.cpp

```diff
@@ -467,52 +519,64 @@ void UxrceddsClient::run()
 
 		while (!should_exit() && _connected) {
 
-			/* Wait for topic updates for max 1000 ms (1sec) */
-			int poll = px4_poll(&_subs->fds[0], (sizeof(_subs->fds) / sizeof(_subs->fds[0])), 1000);
+			perf_begin(_loop_perf);
+			perf_count(_loop_interval_perf);
 
-			/* Handle the poll results */
-			if (poll == 0) {
-				/* Timeout, no updates in selected uorbs */
-				continue;
-
-			} else if (poll < 0) {
-				/* Error */
-				if (poll_error_counter < 10 || poll_error_counter % 50 == 0) {
-					/* Prevent flooding */
-					PX4_ERR("ERROR while polling uorbs: %d", poll);
-				}
+			int orb_poll_timeout_ms = 10;
+
+			int bytes_available = 0;
 
-				poll_error_counter++;
-				continue;
+			if (ioctl(_fd, FIONREAD, (unsigned long)&bytes_available) == OK) {
+				if (bytes_available > 10) {
+					orb_poll_timeout_ms = 0;
```

Comment:
This is mixing up the orb subscription polling with the received data on serial.

Reply:
The idea is to not waste any time sleeping in the (uORB) poll if there's incoming data that can be read and processed.